grails-app/controllers/org/pih/warehouse/partialReceiving/PartialReceivingController.groovy

```diff
@@ -1,17 +1,28 @@
 package org.pih.warehouse.partialReceiving
 
+import org.pih.warehouse.api.StockMovement
+import org.pih.warehouse.core.Location
 import org.pih.warehouse.shipping.Shipment
 
 class PartialReceivingController {
 
     def receiptService
+    def stockMovementService
 
     def index = {
         redirect(action: "create")
     }
 
     def create = {
-        Shipment shipment = Shipment.get(params.id)
+        Location currentLocation = Location.get(session.warehouse.id)
```

Comment:
Remind me to discuss this during our next tech huddle. This was a bad idea (and I kind of knew it would be) especially for the API since we should try to be stateless whenever possible.

Reply:
@jmiranda Sure, I will.